poetry/installation/executor.py

```diff
@@ -261,7 +261,7 @@ def _do_execute_operation(self, operation):
 
             return 0
 
-        if not self._enabled or self._dry_run:
+        if not self._enabled:
```

Comment:
I do not think this is the correct fix. Now, if the operation is a dry run we end up performing the operation since the `return 0` in this block is never hit.

Reply:
Yes, the check should still be there but not write if it is a dry run, sorry for missing that!